rl4co/envs/common/utils.py

```diff
@@ -44,3 +45,56 @@ def _getstate_env(self):
     state = self.__dict__.copy()
     del state["rng"]
     return state
+
+
+class Generator():
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    def __call__(self, batch_size) -> TensorDict:
+        batch_size = [batch_size] if isinstance(batch_size, int) else batch_size
+        return self._generate(batch_size)
+
+    def _generate(self, batch_size, **kwargs) -> TensorDict:
+        raise NotImplementedError
+
+
+def get_sampler(
+        val_name: str,
+        distribution: Union[int, float, str, type, Callable],
+        min_val: float,
+        max_val: float,
+        **kwargs
+    ):
+    """Get the sampler for the variable with the given distribution
+    Args:
+        val_name: Name of the variable
+        distribution: int/float value (as constant distribution), or string with the distribution name (supporting 
+            uniform, normal, exponential, and poisson) or PyTorch Distribution type or a callable function that
+            returns a PyTorch Distribution
+        min_val: Minimum value for the variable, used for Uniform distribution
+        max_val: Maximum value for the variable, used for Uniform distribution
+        kwargs: Additional arguments for the distribution
+    """
+    if isinstance(distribution, (int, float)):
+        return Uniform(low=distribution, high=distribution)
+    elif distribution == "center": # Depot
+        return Uniform(low=(max_val-min_val)/2, high=(max_val-min_val)/2)
+    elif distribution == "corner": # Depot
+        return Uniform(low=min_val, high=min_val)
+    elif distribution == Uniform or distribution == "uniform":
+        return Uniform(low=min_val, high=max_val)
+    elif distribution == Normal or distribution == "normal":
+        assert kwargs.get("mean_"+val_name, None) is not None, "mean is required for Normal distribution"
+        assert kwargs.get(val_name+"_std", None) is not None, "std is required for Normal distribution"
+        return Normal(mean=kwargs[val_name+"_mean"], std=kwargs[val_name+"_std"])
+    elif distribution == Exponential or distribution == "exponential":
+        assert kwargs.get(val_name+"_rate", None) is not None, "rate is required for Exponential/Poisson distribution"
+        return Exponential(rate=kwargs[val_name+"_rate"])
+    elif distribution == Poisson or distribution == "poisson":
+        assert kwargs.get(val_name+"_rate", None) is not None, "rate is required for Exponential/Poisson distribution"
+        return Poisson(rate=kwargs[val_name+"_rate"])
```

Comment:
Do I understand correctly we're assuming a specific format for these parameters then (i.e. we expect parameters `val_name_mean`, `val_name_std`, etc.?) and it's not enough to simply pass f.e. `mean = 5, std = 2`?

Reply:
Good question! Your understand is correct.I have thought about this for some time. The thing is: this `get_sampler()` function will be called in the generator multiple times for different features, e.g. in the `OPGenerator()`

```python
self.depot_sampler = get_sampler("depot", depot_distribution, min_loc, max_loc, **kwargs)
self.prize_sampler = get_sampler("prize", prize_distribution, min_prize, max_prize, **kwargs)
```

If the user wants to init the location with a `Normal` distribution and the prize with a `Poisson` distribution, 3 parameters are required:
1. The mean of location;
2. The std of location;
3. The rate of prize.

We have two options to consider these parameters in the `OPGenerator()`:

1. Adding all of them explicitly to the `__init__()` inputs;
```python
def __init__(self, min_loc, max_loc, mean_loc, std_loc, rate_loc, loc_distribution,\
    min_prize, max_prize, mean_prize, std_prize, rate_prize, prize_distribution)
```
2. Supported by the `kwargs` in the `__init__()` inputs, i.e. the user should follow the rule that: if you want to use the `Normal` distribution for `<val_name>`, you have to give extra parameters with exact the name `mean_<val_name>` and `std_<val_name>`.

Actually, both will work, but for clarity and flexibility, I chose the second way. However, I understand that this would be confusing for users, we should have a clear documentation for the standard rule for the parameter name. 

If you have a better implementation, please tell me 🤔 I do think the current implementation may not be the optimal.